source/backend/dal/Repositories/CompensationRequisitionRepository.cs

```diff
@@ -0,0 +1,91 @@
+using System.Collections.Generic;
+using System.Linq;
+using System.Security.Claims;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.Logging;
+using Pims.Dal.Entities;
+using Pims.Dal.Helpers.Extensions;
+using Pims.Dal.Security;
+
+namespace Pims.Dal.Repositories
+{
+    /// <summary>
+    /// Provides a repository to interact with compensation requisitions within the datasource.
+    /// </summary>
+    public class CompensationRequisitionRepository : BaseRepository<PimsCompensationRequisition>, ICompensationRequisitionRepository
+    {
+        #region Constructors
+
+        /// <summary>
+        /// Creates a new instance of a CompensationRepository, and initializes it with the specified arguments.
+        /// </summary>
+        /// <param name="dbContext"></param>
+        /// <param name="user"></param>
+        /// <param name="logger"></param>
+        public CompensationRequisitionRepository(PimsContext dbContext, ClaimsPrincipal user, ILogger<BaseRepository> logger)
+            : base(dbContext, user, logger)
+        {
+        }
+        #endregion
+
+        public IList<PimsCompensationRequisition> GetAllByAcquisitionFileId(long acquisitionFileId)
+        {
+            return Context.PimsCompensationRequisitions
+                .Include(c => c.PimsCompReqH120s)
+                .AsNoTracking()
+                .Where(c => c.AcquisitionFileId == acquisitionFileId).ToList();
+        }
+
+        public PimsCompensationRequisition Add(PimsCompensationRequisition compensationRequisition)
+        {
+            User.ThrowIfNotAuthorized(Permissions.CompensationRequisitionAdd);
+            Context.PimsCompensationRequisitions.Add(compensationRequisition);
+
+            return compensationRequisition;
+        }
+
+        public void Delete(long compensationRequisitionId)
+        {
+            User.ThrowIfNotAuthorized(Permissions.CompensationRequisitionDelete);
+            var entity = Context.PimsCompensationRequisitions.FirstOrDefault(d => d.CompensationRequisitionId == compensationRequisitionId);
+            if (entity is not null)
+            {
+                Context.PimsCompensationRequisitions.Remove(entity);
```

Comment:
This doesn't appear to remove subentities properly before deleting the main entity, so will likely fail. For example, if a compensation requisition has on or more PimsCompReqH120s, deleting this will cause a foreign key error.

Reply:
will update with the next ticket that handles payees and financial activiites.